### ISSUE TITLE
"Undefined index: distribution_name" for Drupal migrations

### ISSUE THREAD

**Issue by BWPanda** [backdrop/backdrop-issues#4799]
If a Drupal site is upgraded to Backdrop then the installation doesn't run and 'distribution_name' doesn't get set. However when we get that information from the profile info file, we assume it's there and don't allow for instances where it doesn't exist. Then users get the following notice:

> Notice: Undefined index: distribution_name in backdrop_install_profile_distribution_name() (line 222 of [SNIP]\core\includes\install.inc).

We should add a check to backdrop_install_profile_distribution_name() before attempting to return it.

**Comment by BWPanda** [comment]
PR: https://github.com/backdrop/backdrop/pull/3436

**Comment by indigoxela** [comment]
> If a Drupal site is upgraded to Backdrop then the installation doesn't run and 'distribution_name' doesn't get set

This isn't the case for "normal" Drupal installs, I guess, because I wasn't able to reproduce it.

@BWPanda can you provide the steps to reproduce?

**Comment by BWPanda** [comment]
No, I didn't test it, just based my assumptions off the code and a report in the forum...

**Comment by klonos** [comment]
Here's how I tried tested using lando:

- run the following to get myself a fresh installation of latest D7, using the **minimal** installation profile + admin menu: `lando init --source remote --remote-url https://github.com/drupal/drupal/archive/7.x.zip --remote-options="--strip-components 1" --recipe drupal7 --webroot ./drupal-7.x --name d72backdrop && lando start && lando drush si minimal --db-url=mysql://drupal7:drupal7@database:3306/drupal7 -r /app/drupal-7.x -y && drush --root=/app/drupal-7.x dl admin_menu -y && drush --root=/app/drupal-7.x en admin_menu admin_menu_toolbar -y`
- exported the D7 database with: `lando db-export d7.sql`
- copied the db to a safe place
- destroyed the D7 site with: `lando destroy -y`
- removed the D7 codebase
- installed a fresh copy of latest Backdrop, using the **standard** profile: `lando init --source remote --remote-url https://github.com/backdrop/backdrop/archive/1.x.zip --remote-options="--strip-components 1" --recipe backdrop --webroot ./backdrop-1.x --name backdrop && lando start && lando drush si standard --db-url=mysql://backdrop:backdrop@database:3306/backdrop -r /app/backdrop-1.x -y`
- confirmed that the fresh Backdrop site can be accessed, and that I can log in as user1 👍 
- imported the d7 database with: `lando db-import d7.sql.gz`
- tried accessing `/update.php` -> got redirected to `/core/update.php` and a "404 page not found" error

>No, I didn't test it ...

That's where I'm at currently. ...I'm hoping that we can refine the above, so that we have easy to reuse steps to test vanilla D7 to Backdrop upgrades.

**Comment by indigoxela** [comment]
Another test, this time I could reproduce it, because I installed the minimal profile. My steps in detail:

**Drupal:**

- Installed minimal profile (minimal-7.77)
- Enabled a bunch of modules
- Added content types with fields
- Created users, terms, content with devel
- Uninstalled devel
- Made sure, error display is on
- Took database dump

**Backdrop**

- Installed a fresh Backdrop 1.18.x-dev 
- Copied the content of the Drupal files directory to that web
- Dropped and recreated (empty) database of that Backdrop
- Imported the Drupal db dump
- $settings['update_free_access'] = TRUE;
- ran core/update.php (193 pending updates)
- Back to $settings['update_free_access'] = FALSE;

- Got several Undefined index notices
- Enabled Project Installer module (why isn't it enabled by default?)
- Another notice "Notice: Undefined index: distribution_name"
- Ran utf8mb4 conversion
- Changed Public file system path (forgot to do that before)
- Ran cron (updates for images)

Notice: Undefined index: distribution_name still shows up on admin/modules/list
Notice: Undefined index: name/version still shows up on admin/reports/status

Applied the PR patch

Notice: Undefined index: distribution_name is gone
Notice: Undefined index: name/version still shows up on admin/reports/status

**The remaining messages:**

```
    Notice: Undefined index: name in system_requirements() (line 33 of .../core/modules/system/system.install).
    Notice: Undefined index: version in system_requirements() (line 35 of .../core/modules/system/system.install).
```
Besides the notices because it's not the standard profile, everything went smoothly.

**Comment by indigoxela** [comment]
@BWPanda I now reverted the patch and keep this testing web in that state to further test your PR after changes/updates.

Let's see how we make the notices go away. One down, two to go. :wink:

**Comment by BWPanda** [comment]
Thanks @indigoxela! Are those last two notices not part of https://github.com/backdrop/backdrop-issues/issues/2518? Maybe we should fix those there instead if so...

**Comment by indigoxela** [comment]
> Are those last two notices not part of #2518?

Hm, good question, isn't this issue here a duplicate of that older one?

**Comment by BWPanda** [comment]
This is about the distribution name notice, that was about the other two notices.

**Comment by indigoxela** [comment]
> This is about the distribution name notice

Sure, but the root cause for all three notices is the same: profile is not standard. The fact that distribution_name isn't mentioned over there might just have been an oversight.

Your current PR only suppresses the notice, but doesn't solve the base problem. I agree with [that comment](https://github.com/backdrop/backdrop-issues/issues/2518#issuecomment-743071758) by @klonos. We really should handle this better, otherwise upgrading from D7 could cause unnecessary pain.

**Comment by alanmels** [comment]
Please note https://github.com/backdrop-contrib/profile_switcher module had been ported to Backdrop a while ago.

**Comment by jenlampton** [comment]
The underlying problem here is that we're checking the list of modules for a profile, but in the profile's `.info` file we have `type = profile`.

**Comment by jenlampton** [comment]
Well, I just ran the upgrade again and it started enabled, and ended enabled. I don't know what's going on here. But once `minimal` was enabled I didn't get any of the undefined index issues.

**Comment by klonos** [comment]
I'm tentatively adding this to my to-do list ...unless someone else beats me to it.

**Comment by klonos** [comment]
Should we close this as a duplicate of #2518?
